### DISCUSSION TITLE
Low bug discovred

### DISCUSSION THREAD

**Discussion by Tarun5139** [AdnaneKhan/Gato-X#53] [General]
Hello Team , 
Can you help to understand about this 
<img width="1652" alt="Screenshot 2024-10-09 at 9 12 13 PM" src="https://github.com/user-attachments/assets/064b8226-6d4f-445c-a42e-82c59a77cd80">

**Comment by AdnaneKhan** [comment]
Well, this is `workflow_run`. It's only exploitable if the repository has the default workflow approval setting and you are a previous contributor. Take time to understand what is going on. The workflow is downloading an artifact, what is it doing with the file? What is it doing with the output?

You can modify the triggering workflow in your fork to do whatever. Understand how this works and that will quickly reveal what you need to do here.

**Comment by Tarun5139** [comment]
HI @AdnaneKhan , 
Thanks for reply do i need to check with fork 
By by creating pull request and checking approval 
I am wright 

Regards 
Tarun

**Comment by Tarun5139** [comment]
This is workflow of it 


name: Test and lint
permissions: read-all

on:
  pull_request_target:
    branches:
      - main

jobs:
  tests:
    name: Run Tests
    runs-on: ubuntu-latest
    environment: build
    steps:
      - uses: actions/checkout@v3
        with:
          ref: "refs/pull/${{ github.event.number }}/merge"

      - name: Install Foundry
        uses: onbjerg/foundry-toolchain@v1
        with:
          version: nightly

      - name: Install dependencies
        run: forge install
        
      - name: Run tests
        run: forge test --fork-url ${{ secrets.ETH_RPC_URL }} -vv
  lint:
    name: Run Lint
    runs-on: ubuntu-latest
    environment: build
    steps:
      - uses: actions/checkout@v3
        with:
          ref: "refs/pull/${{ github.event.number }}/merge"

      - uses: actions/setup-node@v2
        with:
          node-version: 12.x
      - run: yarn install --frozen-lockfile
      - run: yarn lint:check

**Reply by AdnaneKhan** [reply]
That's an environment protection gated TOCTOU. Your injection point is the `package.json` "lint:check" script.

https://github.com/AdnaneKhan/ActionsTOCTOU

**Reply by AdnaneKhan** [reply]
No impact on that repo by the way, but good learning experience if you mirror it, configure the protection rules yourself and try how it would work.

**Comment by Tarun5139** [comment]
Hello @AdnaneKhan , 
Everytime does relies of what type of branch protection rules are applied on that report

**Comment by AdnaneKhan** [comment]
This is not the appropriate please to ask if something is exploitable or not. Please read about how GitHub Actions works and determine that for yourself.
